consensus/reactor.go

```diff
@@ -430,6 +433,16 @@ func (conR *Reactor) subscribeToBroadcastEvents() {
 		}); err != nil {
 		conR.Logger.Error("Error adding listener for events", "err", err)
 	}
+
+	if conR.conS.config.HasBlockPart {
```

Comment:
I think we might want the if statement inside the callback since the event will always be fired, not sure if we need to ensure theres a listener for it even if its a noop